src/lib/api/cache-purge.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * @flow
+ * @format
+ */
+
+/**
+ * External dependencies
+ */
+import gql from 'graphql-tag';
+
+/**
+ * Internal dependencies
+ */
+import API from 'lib/api';
+
+const mutation = gql`
+	mutation PurgeCacheObjectMutation(
+		$appId: Int!
+		$envId: Int!
+		$urls: [String]!
+	) {
+		purgeCacheObject( input: {
+			appId: $appId
+			environmentId: $envId
+			urls: $urls
+		} ) {
+			status
+			urls
+		}
+	}
+`;
+
+// The subquery for environments lets users choose any environment, including production.
+export const appQuery = `
+	id
+	name
+	environments {
+		id
+		appId
+		name
+		primaryDomain {
+			name
+		}
+		type
+	}
+`;
+
+export async function purgeCache( appId: number, envId: number, urls: Array<string> ) {
+	const api = await API();
+
+	const variables = {
+		appId,
+		envId,
+		urls,
+	};
+
+	const response = await api.mutate( { mutation, variables } );
+
+	const { data: { purgeCacheObject } } = response;
+	if ( ! purgeCacheObject ) {
+		throw new Error( 'Something went wrong :(' );
```

Comment:
Can we be more specific here? Will there be an error returned? Is this even necessary or can we handle the error in the calling code?

Reply:
@chriszarate If the command throws an error, it will be printed on the CLI command, as the domain exception. This error message is just a "backup" if the purgeCacheObject is null and the mutation is not throwing an error.

Reply:
Understood, it's just not particularly useful as an error message. If you removed this check/throw, it would raise an exception whenever you tried to destructure from `null`, which is probably just as useful, if not more so.

If there are situations where we think we might return `null`, we should probably convert them to useful error messages in the API.

Reply:
@chriszarate I agree 👍